src/Snooze/Data.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+module Snooze.Data (
+    Component (..)
+  , Query (..)
+  , Service (..)
+  , Registry (..)
+  , Entry (..)
+  , ContentType (..)
+  ) where
+
+
+import           Control.Retry (RetryPolicy)
+
+import           Data.ByteString (ByteString)
+import           Data.String (IsString)
+import           Data.Text (Text)
+import           Data.Map.Strict (Map)
+
+import           P
+
+import           Snooze.Balance.Data (UpdatableBalanceTable)
+
+newtype Component =
+  Component {
+      component :: Text
```

Comment:
More than happy to wait to see how/where it's used, but not sure what this means without context.